### ISSUE TITLE
Pub can't find code_transformers package versions

### ISSUE THREAD

**Issue by isoos** [dart-lang/pub#1988]
_From @griggsrl on September 27, 2018 19:0_

Original issue details: https://stackoverflow.com/questions/52541532/package-code-transformers-has-no-versions-that-match

I'm not sure what the issue is here with the code_transformers version detection but it appears that the pub query can't find the versions that my dependency needs. Looking at https://pub.dartlang.org/packages/code_transformers/versions the compatible versions do exist out there but may not be returning in the pub query. 

Is this potentially an issue with the index? Or another issue entirely? Any help is greatly appreciated.

Thanks,
Ryan

_Copied from original issue: dart-lang/pub-dartlang-dart#1660_

**Comment by isoos** [comment]
I think the pub site provides the correct version information:
https://pub.dartlang.org/api/packages/code_transformers

Moving this to pub client, as I suspect the issue might be there.

**Comment by jakemac53** [comment]
code_transformers (and barback) are not supported in Dart 2, their sdk constraints will not allow you to get them.

**Comment by jakemac53** [comment]
nevermind I see they are on 1.24

**Comment by natebosch** [comment]
That's a _very_ old `angular` that depends on a _very_ old `code_transformers`.

With `angular` depending on `code_transformers` `0.1.4` you won't be able to use it with that version of pub. I'm not sure the best way to find out the latest version of pub/sdk that allows a `barback` that old...

**Comment by griggsrl** [comment]
Two questions then...

Would simply installing an older package of the Dart SDK be sufficient to find an older version of pub that might be compatible?

Second, it looks like there have been pretty major changes since 0.14 of the angular package. How much of an impact would I have to expect by upgrading to current? Based on what I see in the changelog a lot would have to be essentially redone at this point. Or is the migration not actually as bad as it seems?

**Comment by natebosch** [comment]
> Would simply installing an older package of the Dart SDK be sufficient to find an older version of pub that might be compatible?

Yes it should be, I'm unfortunately not quite sure which version is the latest that would work.

> How much of an impact would I have to expect by upgrading to current? Based on what I see in the changelog a lot would have to be essentially redone at this point. Or is the migration not actually as bad as it seems?

Angular `2.0.0` was a complete rewrite of the framework from `0.14.0`. Breaking changes since then have been a bit smaller, but I'd expect a pretty big change from code written against `0.14.0`. Most of the _concepts_ translate but there were many changes in implementation. Migrating would be less work than a rewrite, but I'd expect to touch code in every component. There are benefits to migrating - the current Angular can help reduce the size of the output.

**Comment by griggsrl** [comment]
Sorry for the delayed response but thanks @natebosch for that info, kind of had a feeling that'd be the case. I went back and installed version 1.12.0 of the SDK which was from around the time that this developer left us. There are no more errors and it compiles successfully so that seems to work now. I'll have to go back and see just how much effort it might take to update to current since that'd obviously be ideal.

Thanks for everyone's help and feedback.
